clang/test/CXX/drs/cwg2xx.cpp

```diff
@@ -649,15 +649,15 @@ namespace cwg241 { // cwg241: 9
     A::g<3>(b);
     C::f<3>(b);
     // expected-error@-1 {{no matching function for call to 'f'}}
-    //   expected-note@#cwg241-C-f {{candidate template ignored: invalid explicitly-specified argument for template parameter 'T'}}
+    //   expected-note@#cwg241-C-f {{candidate template ignored: invalid explicitly-specified argument for template parameter 'T': could not convert '3' from 'int' to class 'T' (expected a class, but got '3')}}
```

Comment:
The newly added diagnostic wording seems confusing to me, but the text in parentheses might be salvageable:
```suggestion
    //   expected-note@#cwg241-C-f {{candidate template ignored: invalid explicitly-specified argument for template parameter 'T': expected a type, but got value '3' (of type 'int'))}}
```

Reply:
Parentheses can be omitted in my suggestion, but the intent is to hammer on type parameter vs value argument mismatch.

Reply:
What you suggested is more like what GCC does, and I have updated it in my last patch, thanks!